api/v1alpha1/releaseplan_types.go

```diff
@@ -17,20 +17,34 @@ limitations under the License.
 package v1alpha1
 
 import (
+	shared "github.com/redhat-appstudio/shared-utils/tekton"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // ReleasePlanSpec defines the desired state of ReleasePlan.
 type ReleasePlanSpec struct {
-	// DisplayName is the long name of the ReleasePlan
-	// +optional
-	DisplayName string `json:"displayName"`
-
 	// Application is a reference to the application to be released in the managed namespace
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +required
 	Application string `json:"application"`
 
+	// Data is an unstructured key used for providing data for the release Pipeline
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	Data *runtime.RawExtension `json:"data,omitempty"`
+
+	// PipelineRef is an optional reference to a Pipeline that would be executed
```

Comment:
This is the first time I hear of this. Where is this defined? I don't see this in the linked Jira. Also, maybe the PR description / commit message could say a bit more about these changes as well?

Reply:
This is what you wrote but just moved to the shared-utils repo.

Reply:
Oh, right. I didn't get your comment. This is a feature that is coming. The code for it will be added with a different ticket but I wanted to do all the API changes on the same PR.

Reply:
ok. but this PR still has no description. could we have at least a brief summary of what is being done here besides just removing ReleaseStrategy?

Reply:
As soon as I fix the commits after an initial review.